pkg/api/validation/validation.go

```diff
@@ -905,8 +905,26 @@ func ValidatePersistentVolumeClaim(pvc *api.PersistentVolumeClaim) field.ErrorLi
 }
 
 func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *api.PersistentVolumeClaim) field.ErrorList {
-	allErrs := field.ErrorList{}
-	allErrs = ValidatePersistentVolumeClaim(newPvc)
+	allErrs := ValidateObjectMetaUpdate(&newPvc.ObjectMeta, &oldPvc.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidatePersistentVolumeClaim(newPvc)...)
+	// if a pvc had a binded volume, we should not allow updates to resources or access modes
+	if len(oldPvc.Spec.VolumeName) != 0 {
+		specPath := field.NewPath("spec")
+		newAccessModes := sets.NewString()
+		for _, accessMode := range newPvc.Spec.AccessModes {
```

Comment:
A helper exists to make a deterministic string from access modes.

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/helpers.go#L203